### PR TITLE
update user-manual

### PR DESCRIPTION
**Pull request by akwei**
update user-manual

### CODE REVIEW

**Comment by zhao-kun** [comment]
User Manual should be concentrating on users how to use our agent instead of how to use EaseAgent, but it isn't enough for this usage, especially fully configuration explanation.

File: doc/user-manual.md [hunk 1]
```diff
@@ -1,18 +1,264 @@
...
+      - [HTTP Request](#http-request)
+      - [JDBC Statement](#jdbc-statement)
+      - [JDBC Connection](#jdbc-connection)
+      - [JVM Memory](#jvm-memory)
+      - [JVM GC](#jvm-gc)
+      - [Kafka Client](#kafka-client)
+      - [RabbitMq Producer](#rabbitmq-producer)
+      - [RabbitMq Consumer](#rabbitmq-consumer)
+  
+## Agent.properties
```
Comment (zhao-kun) [inline doc/user-manual.md:42]: Need a shorten text to describe what's purpose for Agent.properties

File: doc/user-manual.md [hunk 2]
```diff
@@ -1,18 +1,264 @@
...
+---| ---| ---|
+`easeagent.server.enabled` | true | Enable Internal HTTP Server. `False` can disable it. EaseAgent will no longer accept any HTTP requests (`Prometheus`、`Health Check`、`Readiness Check`) when the Internal HTTP Server is disabled. User can add VM parameter:`-Deaseagent.server.enabled=[true or false]` to override.|
+`easeagent.server.port` | 9900 | Internal HTTP Server port. User can add VM parameter:`-Deaseagent.server.port=[new port]` to override. |
+
+### Metric
+Key| Default Value | Description |
+---| ---| ---|
+`observability.metrics.enabled` | true | Enable all metrics collection. `False`: Disable all metrics collection |
+
+### Kafka
```
Comment (zhao-kun) [inline doc/user-manual.md:66]: Why we need to config Kafka address, need a shorten text to answer this question.

File: doc/user-manual.md [hunk 3]
```diff
@@ -1,18 +1,264 @@
...
+
+SubType = "04" // For Producer success
+The producer's Counters metric name for success is [04]+[3]+[demo-topic] = 043demo-topic
+ 
+SubType = "06" // For Producer error
+The producer's Counters metric key for error is [06]+[3]+[demo-topic] = 063demo-topic 
+```
+**The metric key is `SubType`+`Type`+`SubKey`**
+
+#### Type And SubType Detail
```
Comment (zhao-kun) [inline doc/user-manual.md:261]: I think this section should belong to development guide.

File: doc/user-manual.md [hunk 4]
```diff
@@ -1,18 +1,264 @@
...
+rabbit.queue | RabbitMQ routingKey |
+
+#### Kafka Producer And Consumer
+Tag | Description |
+--- | ---|
+kafka.key | Kafka consumer record Key |
+kafka.topic | Kafka topic |
+kafka.broker | Kafka url |
+
+## Metric
```
Comment (zhao-kun) [inline doc/user-manual.md:241]: I need to tell users that we pack the metrics value in what's packge format/protocol

**Review by zhao-kun — COMMENTED** [review]
You must assume the users have no background knowledge about EaseAgent. I think the current manual is prepare for us instead of them.

File: doc/user-manual.md (hunk 4 above)
Comment (akwei) [reply doc/user-manual.md]: > We need to tell users that we pack the metrics value in what's packge format/protocol

I will move Metric Info to `Developement-guide` and fix it

File: doc/user-manual.md (hunk 1 above)
Comment (akwei) [reply doc/user-manual.md:41]: fixed

File: doc/user-manual.md [hunk 5]
```diff
@@ -1,216 +1,457 @@
...
+      - [JVM GC](#jvm-gc)
+      - [Kafka Client](#kafka-client)
+      - [RabbitMq Producer](#rabbitmq-producer)
+      - [RabbitMq Consumer](#rabbitmq-consumer)
+  
+## Agent.properties
+`EaseAgent` uses the `report` module to control the tracing and metric collection behavior of each component. `agent.properties` is configured with various parameters of `report`. Changing these parameters can change the collection behavior. These parameters include: collection frequency, target queue, switch and other settings. Users can customize parameter.
+
+### Getting the configuration file
+Extracting the default configuration file
```
Comment (zhao-kun) [inline doc/user-manual.md:45]: you may extract default configuration from the JAR file or create new properties from a blank file

File: doc/user-manual.md [hunk 6]
```diff
@@ -1,216 +1,457 @@
...
+  "p50" : 823,
+  "host_ipv4" : "192.168.2.5",
+  "m1errpct" : 0,
+  "m15errpct" : 0,
+  "m1err" : 0,
+  "p999" : 823
+}
+```
+
+For different components, the fields contained in JSON are as follows:
```
Comment (zhao-kun) [inline doc/user-manual.md:290]: For different kind of metrics, we have different schemas:

File: doc/user-manual.md [hunk 7]
```diff
@@ -1,216 +1,457 @@
...
+
+#### HTTP Request
```
Comment (zhao-kun) [inline doc/user-manual.md:292]: HTTP Request schema describes key metrics of service APIs, which include:
- Throughput (M1, M5, M15),
- Error throughput (ERROR M1, M5, M15)
- Latency (p99, p98,p95, p75,p50...)

**Review by zhao-kun — COMMENTED** [review]
Give each schema a text to describe what's contents of it

File: doc/user-manual.md (hunk 7 above)
Comment (akwei) [reply doc/user-manual.md:298]: I fixed, and remove some unused fields

**Review by zhao-kun — APPROVED** [review]
LGTM

**Pull request merged**